aptos-move/move-examples/fungible_asset/stablecoin/sources/fiat_token.move

```diff
@@ -0,0 +1,420 @@
+            &metadata_object_signer,
+            FiatTokenConfig { mint_ref, burn_ref, transfer_ref },
+        );
+        // store access info in `FiatTokenAccessInfo` resource, move resource to `Circle/Owner` address
+        move_to(
+            owner,
+            FiatTokenAccessInfo {
+                minters: smart_table::new(),
+                blacklist: vector::empty(),
+            },
+        );
+        // store master minter in `MasterMinter` resource, move resource to `Circle/Owner` address
+        move_to(
+            owner,
+            MasterMinterConfig { master_minter },
+        )
+    }
+    
+    /// Mint new fiat tokens and deposit them to the `to` address.
+    /// Returns true if operation is successful.
+    public entry fun mint(minter: &signer, to: address, amount: u64) acquires FiatTokenConfig, FiatTokenAccessInfo  {
+        // SECURITY TODO: check if contract is paused 
+        // SECURITY TODO: check if transaction authorizer (signer) is a minter OR owner
+
+        // check if `minter` or `to` address is blacklisted
+        assert!(!is_blacklisted(signer::address_of(minter)), error::permission_denied(EBLACKLISTED_ADDRESS));
+        assert!(!is_blacklisted(to), error::permission_denied(EBLACKLISTED_ADDRESS));
+        
+        // check if amount is greater than the minter's allowance 
+        assert!(get_minter_allowance(signer::address_of(minter)) > amount, error::permission_denied(EMINTER_ALLOWANCE_EXCEEDED));
+        
+        // retrieve the refs using the `Metadata` object
+        let metadata_object = get_metadata_object();
+        let fiat_token_config = authorized_borrow_refs(minter, metadata_object);
+
+        // mint the tokens and deposit the minted tokens to the `to` address
+        let to_wallet = primary_fungible_store::ensure_primary_store_exists(to, metadata_object);
+        let fiat_token = fungible_asset::mint(&fiat_token_config.mint_ref, amount);
+        fungible_asset::deposit_with_ref(&fiat_token_config.transfer_ref, to_wallet, fiat_token);
+
+        // decrease minter allowance by amount
+        decrement_minter_allowance(signer::address_of(minter), amount);
+
+        // TODO: emit minting event (??) or is FA events sufficient, indexing could be done that way
+    }
+
+    /// Burn fiat tokens from the minter's primary store.
+    /// Returns true if operation is successful.
+    public entry fun burn(minter: &signer, amount: u64) acquires FiatTokenConfig, FiatTokenAccessInfo {
+        // SECURITY TODO: check if contract is paused 
+        // SECURITY TODO: check if transaction authorizer (signer) is a minter OR owner
+
+        // check if `minter` is blacklisted
+        assert!(!is_blacklisted(signer::address_of(minter)), error::permission_denied(EBLACKLISTED_ADDRESS));
+
+        // SECURITY TODO: check minter balance >= amount
+        assert!(balance_of_primary_store(signer::address_of(minter)) < amount, error::permission_denied(EINSUFFICIENT_BALANCE));
```

Comment:
Similarly, I think permission_denied is the wrong one for `INSUFFICIENT_BALANCE`.

Also, needs to be balance >= amount.

Right now it'll only pass if the amount in the store is less than the burn amount.